### PR TITLE
implemented dropping user defined function

### PR DESCRIPTION
**Pull request by bogensberger**

### CODE REVIEW

File: sql/src/main/java/io/crate/operation/udf/TransportUserDefinedFunctionResponse.java [hunk 1]
```diff
@@ -32,12 +32,12 @@
 
 import java.io.IOException;
 
-class CreateUserDefinedFunctionResponse extends AcknowledgedResponse {
+class TransportUserDefinedFunctionResponse extends AcknowledgedResponse {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/udf/TransportUserDefinedFunctionResponse.java]: Why the rename? We usually don't include `Transport` in request/response names

File: sql/src/main/java/io/crate/operation/udf/UserDefinedFunctionService.java [hunk 2]
```diff
@@ -109,6 +112,36 @@ protected ClusterStateUpdateResponse newResponse(boolean acknowledged) {
             });
     }
 
+    void dropFunction(final DropUserDefinedFunctionRequest request,
+                      final ActionListener<ClusterStateUpdateResponse> listener) {
+        if (!clusterService.localNode().isMasterNode()) {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/udf/UserDefinedFunctionService.java]: This is only checking if this node *could* be the master, not if it currently *is* the master.

Need to use something like this:

```
        DiscoveryNodes nodes = state.nodes();
        if (nodes.getMasterNodeId().equals(nodes.getLocalNodeId()) == false) {
```

File: sql/src/main/java/io/crate/operation/scalar/arithmetic/AddFunction.java [hunk 3]
```diff
@@ -84,7 +84,7 @@ public Number evaluate(Input[] args) {
         }
     }
 
-    private static class LongAddFunction extends AddFunction {
+    public static class LongAddFunction extends AddFunction {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/operation/scalar/arithmetic/AddFunction.java]: why the change here?

Comment (kovrus) [inline sql/src/main/java/io/crate/operation/scalar/arithmetic/AddFunction.java]: mistake...
